test/correctness/register_shuffle.cpp

```diff
@@ -542,9 +542,9 @@ int main(int argc, char **argv) {
     {
         // Test a case that caused combinatorial explosion
         Var x;
-        Expr e = x;
+        Expr e = cast<float>(x);
```

Comment:
Why is this cast necessary? Shouldn't fast_pow do it?

Reply:
I changed the code in fast-pow to not do that. This technically an API-breaking change. The good news is that if you do pass an int (like this test did), it will just not compile (so no implicit behavior change (besides the actual approximations being re-implemented of course)). Passing an `int` in fast_pow seems ambiguous. Do you want single, double, or half precision results? Keep in mind that one of my design goals was that this new API would not prevent us from expanding this to float16 (and float64??) arguments/evaluations at some later point. Fast_pow was the only function that internally was doing an implicit cast in main right now.

Reply:
Reasonable, but because it's API-breaking, how about leaving the cast in fast_pow for now, but adding a deprecation user_warning if the type is anything other than Float(32) so that we can delete it after the next release